src/main/kotlin/com/colivery/serviceaping/security/FirebaseAuthenticationConverter.kt

```diff
@@ -0,0 +1,34 @@
+package com.colivery.serviceaping.security
+
+import com.google.firebase.auth.FirebaseAuth
+import org.springframework.http.HttpHeaders
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.Authentication
+import org.springframework.security.web.server.authentication.ServerAuthenticationConverter
+import org.springframework.web.server.ServerWebExchange
+import reactor.core.publisher.Mono
+
+class FirebaseAuthenticationConverter : ServerAuthenticationConverter {
+
+    override fun convert(exchange: ServerWebExchange): Mono<Authentication> {
+
+        val token: String? = exchange.request.headers.getFirst(HttpHeaders.AUTHORIZATION)?.let { getToken(it) }
+
+        return when (val firebaseUser = token?.let { getFirebaseUser(it) }) {
+            null -> Mono.empty()
+            else -> Mono.just(UsernamePasswordAuthenticationToken(firebaseUser.name, token))
```

Comment:
Wäre es hier nicht noch besser, wenn wir unsere eigene UserEntity in den Auth Context hauen, anstatt nur den Firebase Usernamen? So hätten wir die jederzeit griffbereit.